torch/package/exporter.py

```diff
@@ -69,12 +70,15 @@ def __init__(self, f: Union[str, Path, BinaryIO], verbose: bool = True):
         """
         if isinstance(f, (Path, str)):
             f = str(f)
+            self.name = f
             self.buffer: Optional[BinaryIO] = None
         else:  # is a byte buffer
+            self.name = '<binary>'
             self.buffer = f
 
         self.zip_file = torch._C.PyTorchFileWriter(f)
         self.zip_file.set_min_version(6)
+        self._written_records : List[str] = []
```

Comment:
This approach duplicates the list of records here and in the zip file. It's not a great idea since eventually this will share a zip archive with the JIT which won't use the python API for putting the stuff in the archive. So this would miss those files.

Reply:
self.name is similar: the zip file knows it already, so duplicating it would make it harder.

Reply:
It should be possible for the zip file object to track the files it has written so that it can also implement `get_all_records` similar to the reader. The zip file already has an archiveName function, maybe just needs to be bound into Python.

Reply:
I moved the generation of the records list functionality to the PythonStreamWriter in CPP, it's now in the PR below this one 👍